lib/mayaUsd/fileio/translators/translatorMayaReference.cpp

```diff
@@ -424,29 +472,74 @@ MStatus UsdMayaTranslatorMayaReference::update(const UsdPrim& prim, MObject pare
     // to be a common thing in the workflow.)
     //
     if (refNode.isNull()) {
-        for (MItDependencyNodes refIter(MFn::kReference); !refIter.isDone(); refIter.next()) {
-            MObject      tempRefNode = refIter.item();
-            MFnReference tempRefFn(tempRefNode);
-            if (!tempRefFn.isFromReferencedFile()) {
-
-                // Get the name of the reference node so we can check if this node
-                // matches the prim we are processing.  Strip off the "_RN" suffix.
-                //
-                MString refNodeName = tempRefFn.name();
-                MString refName(refNodeName.asChar(), refNodeName.numChars() - 3);
-
-                // What reference node name is the prim expecting?
-                MString expectedRefName = refNameFromPath(parentDag);
-
-                // If found a match, reconnect the reference node's `associatedNode`
-                // attr before loading it, since the previous connection may be gone.
-                //
-                if (refName == expectedRefName) {
-                    // Reconnect the reference node's `associatedNode` attr before
-                    // loading it, since the previous connection may be gone.
-                    connectReferenceAssociatedNode(parentDag, tempRefFn);
-                    refNode = tempRefNode;
-                    break;
+        if (isMayaReference(prim)) {
+            // New behaviour for MayaReference
```

Comment:
Do you think you could adapt the test I made in #1620 to run on this code? Depending on exactly how you envision this working, all you would need is to update the stage file with the ref node name and see that it reconnects properly when the prim path changes.

Reply:
awesome thanks for including it.